### PR TITLE
Add INT8 type

### PR DESCRIPTION
**Pull request by Ashleyhx**
Completing the numerical typing system - add INT8 dataType, so that entries of type INT8 can be stored into the table. 
An example can be found in StudyAt.csv.

### CODE REVIEW

File: Testing/Temporary/CTestCostData.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+---
```
Comment (acquamarin) [inline Testing/Temporary/CTestCostData.txt:1]: revert

File: Testing/Temporary/LastTest.log [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Start testing: Sep 05 16:10 EDT
```
Comment (acquamarin) [inline Testing/Temporary/LastTest.log:1]: ditto, revert this file.

File: src/c_api/value.cpp [hunk 3]
```diff
@@ -185,6 +185,10 @@ bool kuzu_value_get_bool(kuzu_value* value) {
     return static_cast<Value*>(value->_value)->getValue<bool>();
 }
 
+int8_t kuzu_value_get_int8(kuzu_value* value) {
```
Comment (acquamarin) [inline src/c_api/value.cpp:188]: For this PR, we don't need to add int8_t support for other apis.

File: src/include/common/types/types.h [hunk 4]
```diff
@@ -78,6 +78,8 @@ KUZU_API enum class LogicalTypeID : uint8_t {
     INT64 = 23,
     INT32 = 24,
     INT16 = 25,
+    INT8 = 32,
```
Comment (acquamarin) [inline src/include/common/types/types.h:81]: Can you make INT8 ID to 26?

File: src/include/common/types/types.h [hunk 5]
```diff
@@ -105,6 +107,7 @@ enum class PhysicalTypeID : uint8_t {
     INT64 = 2,
     INT32 = 3,
     INT16 = 4,
+    INT8 = 11,
```
Comment (acquamarin) [inline src/include/common/types/types.h:110]: Make INT8 ID to 5, and increse DOUBLE->6, FLOAT->7...

File: src/include/function/cast/cast_functions.h [hunk 6]
```diff
@@ -157,6 +157,13 @@ inline void CastToInt32::operation(int64_t& input, int32_t& result) {
     numericDownCast<int64_t, int32_t>(input, result, "INT32");
 }
 
+struct CastToInt8 {
```
Comment (acquamarin) [inline src/include/function/cast/cast_functions.h:160]: Let's implement the casting/functions for int8 in the next PR.

**Review by acquamarin — COMMENTED** [review]
You may need to change the ValueVector,factorizedTable classes(which are used in query processing), so we can store the INT8 value in these internal data structures.
To add a test in kuzu:
We have a small dataset that used for testing: tinysnb (dataset/tinysnb). You can add a INT8 column in the vMovies.csv file. We usually write simple projection tests in this file: test/test_files/tinysnb/projection/single_label.test

File: src/common/arrow/arrow_row_batch.cpp [hunk 7]
```diff
@@ -274,6 +277,9 @@ void ArrowRowBatch::copyNonNullValue(
     case LogicalTypeID::INT16: {
         templateCopyNonNullValue<LogicalTypeID::INT16>(vector, typeInfo, value, pos);
     } break;
+    case LogicalTypeID::INT8: {
+        templateCopyNonNullValue<LogicalTypeID::INT8>(vector, typeInfo, value, pos);
+    } break;
```
Comment (acquamarin) [inline src/common/arrow/arrow_row_batch.cpp:282]: You can remove the case statements in this file.

File: src/common/types/types.cpp [hunk 8]
```diff
@@ -22,6 +22,8 @@ std::string PhysicalTypeUtils::physicalTypeToString(PhysicalTypeID physicalType)
         return "INT32";
     case PhysicalTypeID::INT16:
         return "INT16";
+    case PhysicalTypeID::INT8:
+        return "INT8";
```
Comment (acquamarin) [inline src/common/types/types.cpp:26]: This is for exception handling, keep this.

File: src/common/types/value.cpp [hunk 9]
```diff
@@ -502,6 +517,9 @@ void Value::serialize(FileInfo* fileInfo, uint64_t& offset) const {
     case PhysicalTypeID::INT16: {
         SerDeser::serializeValue(val.int16Val, fileInfo, offset);
     } break;
+    case PhysicalTypeID::INT8: {
+        SerDeser::serializeValue(val.int8Val, fileInfo, offset);
```
Comment (acquamarin) [inline src/common/types/value.cpp:521]: Keep switch cases in this file, they are used for literalExpression in UDF. You don't need to add a test for this.

File: src/common/vector/value_vector.cpp [hunk 10]
```diff
@@ -140,6 +140,9 @@ void ValueVector::copyFromValue(uint64_t pos, const Value& value) {
     case PhysicalTypeID::INT16: {
         memcpy(dstValue, &value.val.int16Val, numBytesPerValue);
     } break;
+    case PhysicalTypeID::INT8: {
+        memcpy(dstValue, &value.val.int8Val, numBytesPerValue);
```
Comment (acquamarin) [inline src/common/vector/value_vector.cpp:144]: Keep case for int8 in this file.

File: src/include/common/types/value.h [hunk 11]
```diff
@@ -463,6 +475,15 @@ inline bool& Value::getValueReference() {
     return val.booleanVal;
 }
 
+/**
+ * @return the reference to the int8 value.
+ */
+KUZU_API template<>
+inline int8_t& Value::getValueReference() {
+    assert(dataType->getLogicalTypeID() == LogicalTypeID::INT8);
+    return val.int8Val;
```
Comment (acquamarin) [inline src/include/common/types/value.h:484]: keep this one.

File: src/processor/operator/order_by/order_by_key_encoder.cpp [hunk 12]
```diff
@@ -216,6 +216,10 @@ void OrderByKeyEncoder::getEncodingFunction(PhysicalTypeID physicalType, encode_
         func = encodeTemplate<int16_t>;
         return;
     }
+    case PhysicalTypeID::INT8: {
+        func = encodeTemplate<int8_t>;
+        return;
```
Comment (acquamarin) [inline src/processor/operator/order_by/order_by_key_encoder.cpp:221]: Add a test which order by a int8 column under  `order_by/single_label.test`

File: src/processor/operator/persistent/csv_file_writer.cpp [hunk 13]
```diff
@@ -77,6 +77,8 @@ void CSVFileWriter::writeValue(ValueVector* vector) {
         return writeToBuffer<int32_t>(vector, selPos);
     case LogicalTypeID::INT16:
         return writeToBuffer<int16_t>(vector, selPos);
+    case LogicalTypeID::INT8:
+        return writeToBuffer<int8_t>(vector, selPos);
```
Comment (acquamarin) [inline src/processor/operator/persistent/csv_file_writer.cpp:81]: remove

File: src/storage/copier/column_chunk.cpp [hunk 14]
```diff
@@ -431,6 +438,9 @@ void FixedListColumnChunk::write(const Value& fixedListVal, uint64_t posToWrite)
         case PhysicalTypeID::INT16: {
             memcpy(bufferToWrite, &val->getValueReference<int16_t>(), numBytesPerValueInList);
         } break;
+        case PhysicalTypeID::INT8: {
+            memcpy(bufferToWrite, &val->getValueReference<int8_t>(), numBytesPerValueInList);
```
Comment (acquamarin) [inline src/storage/copier/column_chunk.cpp:442]: remove

File: src/storage/copier/npy_reader.cpp [hunk 15]
```diff
@@ -162,6 +162,8 @@ void NpyReader::parseType(std::string descr) {
         type = LogicalTypeID::INT32;
     } else if (descr == "i2") {
         type = LogicalTypeID::INT16;
+    } else if (descr == "i1") {
+        type = LogicalTypeID::INT8;
```
Comment (acquamarin) [inline src/storage/copier/npy_reader.cpp:166]: remove all int8 related code in this file

File: src/storage/copier/struct_column_chunk.cpp [hunk 16]
```diff
@@ -117,6 +117,10 @@ void StructColumnChunk::setValueToStructField(
         fieldChunk->setValueFromString<int16_t>(
             structFieldValue.c_str(), structFieldValue.length(), pos);
     } break;
+    case LogicalTypeID::INT8: {
+        fieldChunk->setValueFromString<int8_t>(
```
Comment (acquamarin) [inline src/storage/copier/struct_column_chunk.cpp:121]: modify a csv file under tinysnb so it contains an int8_t field. This code path is used to support copying a int8_t filed in struct datatype.

File: src/storage/copier/table_copy_utils.cpp [hunk 17]
```diff
@@ -283,6 +284,11 @@ std::unique_ptr<uint8_t[]> TableCopyUtils::getArrowFixedList(const std::string&
             memcpy(listVal.get() + numElementsRead * sizeof(int16_t), &val, sizeof(int16_t));
             numElementsRead++;
         } break;
+        case LogicalTypeID::INT8: {
+            auto val = StringCastUtils::castToNum<int8_t>(element.c_str(), element.length());
+            memcpy(listVal.get() + numElementsRead * sizeof(int8_t), &val, sizeof(int8_t));
+            numElementsRead++;
+        } break;
```
Comment (acquamarin) [inline src/storage/copier/table_copy_utils.cpp:291]: remove

File: src/storage/copier/table_copy_utils.cpp [hunk 18]
```diff
@@ -370,6 +379,10 @@ bool TableCopyUtils::tryCast(
         int16_t result;
         return StringCastUtils::tryCastToNum(value, length, result);
     }
+    case LogicalTypeID::INT8: {
```
Comment (acquamarin) [inline src/storage/copier/table_copy_utils.cpp:382]: Keep this switch. Will be used by union datatype. You don't need to add a test for now

File: package.json [hunk 19]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (acquamarin) [inline package.json:1]: Remove this file

File: package-lock.json [hunk 20]
```diff
@@ -0,0 +1,47 @@
+{
```
Comment (acquamarin) [inline package-lock.json:1]: remove

File: src/include/common/type_utils.h [hunk 21]
```diff
@@ -23,7 +23,14 @@ class StringCastUtils {
         if (iss.str().empty()) {
             throw ConversionException{"Empty string."};
         }
-        iss >> result;
+
+        if constexpr (std::is_same_v<int8_t, T>) {
```
Comment (acquamarin) [inline src/include/common/type_utils.h:27]: Why do you need a if statement here?  `iss >> result` should convert the string to an int8 value.

File: src/include/common/types/types.h [hunk 22]
```diff
@@ -78,12 +78,14 @@ KUZU_API enum class LogicalTypeID : uint8_t {
...
     INT32 = 24,
     INT16 = 25,
-    DOUBLE = 26,
-    FLOAT = 27,
-    DATE = 28,
-    TIMESTAMP = 29,
-    INTERVAL = 30,
-    FIXED_LIST = 31,
+    INT8 = 26,
+
```
Comment (acquamarin) [inline src/include/common/types/types.h:82]: remove this empty line

File: test/graph_test/graph_test.cpp [hunk 23]
```diff
@@ -153,7 +153,7 @@ void TestHelper::executeScript(const std::string& cypherScript, Connection& conn
         std::cout << "Executed query: " << line << std::endl;
         if (!result->isSuccess()) {
             throw Exception(StringUtils::string_format(
-                "Failed to execute statement: {}.\nError: {}", line, result->getErrorMessage()));
+                "Failed to execute statement: {}.\nError: {}", line, result->getErrorMessage())); //
```
Comment (acquamarin) [inline test/graph_test/graph_test.cpp:156]: Delete '//'

**Review by acquamarin — APPROVED** [review]
Remember to increase the KUZU_VERSION in CMAKE_LISTS.TXT, and `src/include/storage/storage_info.h`.

File: src/include/common/type_utils.h (hunk 21 above)
Comment (Ashleyhx) [reply src/include/common/type_utils.h:27]: Because >> treats int8_t as an unsigned char type in C++, it will result in iss.failed() if we don't do this.

**Pull request merged**